plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java

```diff
@@ -110,18 +113,36 @@ private T tryConstructor(Constructor<T> constructor, Mapping config, Configurati
 
             for (int i = 0; i < names.length; i++) {
                 final CNode value = config.get(names[i]);
-                if (value == null && parameters[i].getAnnotation(Nonnull.class) != null) {
-                    throw new ConfiguratorException(names[i] + " is required to configure " + target);
-                }
                 final Class t = parameters[i].getType();
+
+                if (value == null) {
+                    Class<?> clazz = constructor.getDeclaringClass();
+                    if (parameters[i].isAnnotationPresent(Required.class)) {
```

Comment:
I believe `org.springframework.beans.factory.annotation.Required` comes from Jenkins core and could be useful to mark parameters that are explicitly needed for configuration.

Reply:
I've never seen anyone use Required before =/
Nonnull and nullable are the normal ones.

Reply:
JCasC plugin already takes nullability into account in some cases, and IMHO this is the only way to get the things working without updating existing plugins. Also, Jenkins uses a very old version of Spring inside the core, and usage of its contents is not advised taking potential migration issues in the future

Reply:
Using `@Nonnull` should mean that the parameter we pass is not `null`; not that it is 'required'. If that's fine, we can remove the thrown exception in this case. What would you suggest here?